osquery/tables/networking/posix/interfaces.cpp

```diff
@@ -187,7 +219,28 @@ void genDetailsFromAddr(const struct ifaddrs* addr, QueryData& results) {
     r["odrops"] = INTEGER(0);
     r["collisions"] = BIGINT_FROM_UINT32(ifd->ifi_collisions);
     r["last_change"] = BIGINT_FROM_UINT32(ifd->ifi_lastchange.tv_sec);
-#endif
+
+    int fd = socket(AF_INET, SOCK_DGRAM, 0);
```

Comment:
Plumb the `QueryContext& context` parameter through from the caller, and then wrap all this new code in `if (context.isColumnUsed("link_speed")`